### DISCUSSION TITLE
Process TanDEM-X with ISCE2

### DISCUSSION THREAD

**Discussion by ES20180816** [isce-framework/isce2#404] [Q&A]
Hello everyone,

I am trying to process a group of bistastic TanDEM-X images according to the insarApp.py.

However, when I run the code of "make_raw", there is one error as "iscesys.Component.PortError: 'No input port named sensor",  as shown below

>>> def setTDX(info):
...     objSensor = createSensor('TANDEMX')
...     objSensor.xml = getInfo(info, 'xml')
...     objSensor.output = getInfo(info, 'output')
...     # set Doppler
...     objDop = createDoppler('usedopIQ')
...     #dict_keys(['USEDOPIQ', 'USECALCDOP', 'USEDOPPLER', 'USEDEFAULT'])
...     return objSensor, objDop
... 
>>> def make_raw(sensor, doppler):
...     from make_raw import make_raw
...     objMakeRaw = make_raw()
...     objMakeRaw(sensor=sensor, doppler=doppler)
...     return objMakeRaw
... 
>>> (objSensor, objDop) = setTDX(masterXml)
>>> sarMeta = make_raw(objSensor, objDop)
2021-11-28 10:13:36,142 - isce.make_raw - INFO - Extracting image
Number of bands:  1
Driver:  COSAR / COSAR Annotated Binary Matrix (TerraSAR-X)
Size is  9566 x 15030
Band Type= CInt16
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 4, in make_raw
  File "/home/*/anaconda3/envs/isce2/lib/python3.8/site-packages/isce/components/iscesys/Component/Component.py", line 179, in __call__
    return getattr(self, self.__class__.__name__.lower())(*args)
  File "/media/*/a156688b-c172-4ff1-8b73-74597a82bc0b/test_ISCE/TDX/make_raw.py", line 281, in make_raw
    self.doppler.wireInputPort(name='sensor', object=self.sensor)
  File "/home/*/anaconda3/envs/isce2/lib/python3.8/site-packages/isce/components/iscesys/Component/Component.py", line 157, in method_with_flow
    return method(self)(*args, **kwargs)
  File "/home/*/anaconda3/envs/isce2/lib/python3.8/site-packages/isce/components/iscesys/Component/Component.py", line 259, in _wirePort
    raise PortError("No %s port named %s" % (port_iterator.flow, name))
iscesys.Component.Component.PortError: 'No input port named sensor'
>>> 

If anyone can give some suggestions for solving this problem I would be very appreciative.

**Comment by fdelgadodelapuente** [comment]
The TanDEM-X CoSSC processing is not implemented in either the stack processor nor stripmapApp

**Reply by ES20180816** [reply]
Ok, thank you.

**Comment by piyushrpt** [comment]
There are no prepackaged workflows for bistatic data but there are a couple of modules that can be used with python scripts to do the custom processing. 

https://github.com/isce-framework/isce2/tree/main/components/zerodop/bistaticgeo2rdr
https://www.researchgate.net/profile/Yang-Lei-64/publication/349432065_Tropical_Forest_Height_and_Underlying_Topography_from_Tandem-X_SAR_Interferometry/links/605b87f9299bf17367685a94/Tropical-Forest-Height-and-Underlying-Topography-from-Tandem-X-SAR-Interferometry.pdf

**Reply by ES20180816** [reply]
Ok, I have got it. Thank you very much.
